lemmy_db/src/post_report.rs

```diff
@@ -0,0 +1,195 @@
+use diesel::{dsl::*, pg::Pg, result::Error, *};
+use serde::{Deserialize, Serialize};
+
+use crate::{limit_and_offset, MaybeOptional, schema::post_report, post::Post, Reportable, naive_now};
+
+table! {
+    post_report_view (id) {
+        id -> Int4,
+        creator_id -> Int4,
+        post_id -> Int4,
+        post_name -> Varchar,
+        post_url -> Nullable<Text>,
+        post_body -> Nullable<Text>,
+        reason -> Text,
+        resolved -> Bool,
+        resolver_id -> Nullable<Int4>,
+        published -> Timestamp,
+        updated -> Nullable<Timestamp>,
+        current_post_name -> Varchar,
+        current_post_url -> Nullable<Text>,
+        current_post_body -> Nullable<Text>,
+        community_id -> Int4,
+        creator_name -> Varchar,
+        post_creator_id -> Int4,
+        post_creator_name -> Varchar,
+    }
+}
+
+#[derive(Identifiable, Queryable, Associations, PartialEq, Serialize, Deserialize, Debug)]
+#[belongs_to(Post)]
+#[table_name = "post_report"]
+pub struct PostReport {
+  pub id: i32,
+  pub creator_id: i32,
+  pub post_id: i32,
+  pub post_name: String,
+  pub post_url: Option<String>,
+  pub post_body: Option<String>,
+  pub reason: String,
+  pub resolved: bool,
+  pub resolver_id: Option<i32>,
+  pub published: chrono::NaiveDateTime,
+  pub updated: Option<chrono::NaiveDateTime>,
+}
+
+#[derive(Insertable, AsChangeset, Clone)]
+#[table_name = "post_report"]
+pub struct PostReportForm {
+  pub creator_id: i32,
+  pub post_id: i32,
+  pub post_name: String,
+  pub post_url: Option<String>,
+  pub post_body: Option<String>,
```

Comment:
Ah, I see these are the old attributes of the post, and you join to get the current ones below too.

Reply:
Might be better to name them `original_post_body` or similar, to avoid confusion with the current post body (and similar for url, title, and comment fields).